cmd/server/app/options/options_test.go

```diff
@@ -0,0 +1,111 @@
+	assertDefaultValue(t, "ServerKey", defaultServerOptions.ServerKey, "")
+	assertDefaultValue(t, "ServerCaCert", defaultServerOptions.ServerCaCert, "")
+	assertDefaultValue(t, "ClusterCert", defaultServerOptions.ClusterCert, "")
+	assertDefaultValue(t, "ClusterKey", defaultServerOptions.ClusterKey, "")
+	assertDefaultValue(t, "ClusterCaCert", defaultServerOptions.ClusterCaCert, "")
+	assertDefaultValue(t, "Mode", defaultServerOptions.Mode, "grpc")
+	assertDefaultValue(t, "UdsName", defaultServerOptions.UdsName, "")
+	assertDefaultValue(t, "DeleteUDSFile", defaultServerOptions.DeleteUDSFile, false)
+	assertDefaultValue(t, "ServerPort", defaultServerOptions.ServerPort, 8090)
+	assertDefaultValue(t, "ServerBindAddress", defaultServerOptions.ServerBindAddress, "")
+	assertDefaultValue(t, "AgentPort", defaultServerOptions.AgentPort, 8091)
+	assertDefaultValue(t, "AgentBindAddress", defaultServerOptions.AgentBindAddress, "")
+	assertDefaultValue(t, "HealthPort", defaultServerOptions.HealthPort, 8092)
+	assertDefaultValue(t, "HealthBindAddress", defaultServerOptions.HealthBindAddress, "")
+	assertDefaultValue(t, "AdminPort", defaultServerOptions.AdminPort, 8095)
+	assertDefaultValue(t, "AdminBindAddress", defaultServerOptions.AdminBindAddress, "127.0.0.1")
+	assertDefaultValue(t, "KeepaliveTime", defaultServerOptions.KeepaliveTime, 1*time.Hour)
+	assertDefaultValue(t, "FrontendKeepaliveTime", defaultServerOptions.FrontendKeepaliveTime, 1*time.Hour)
+	assertDefaultValue(t, "EnableProfiling", defaultServerOptions.EnableProfiling, false)
+	assertDefaultValue(t, "EnableContentionProfiling", defaultServerOptions.EnableContentionProfiling, false)
+	assertDefaultValue(t, "ServerCount", defaultServerOptions.ServerCount, uint(1))
+	assertDefaultValue(t, "AgentNamespace", defaultServerOptions.AgentNamespace, "")
+	assertDefaultValue(t, "AgentServiceAccount", defaultServerOptions.AgentServiceAccount, "")
+	assertDefaultValue(t, "KubeconfigPath", defaultServerOptions.KubeconfigPath, "")
+	assertDefaultValue(t, "KubeconfigQPS", defaultServerOptions.KubeconfigQPS, float32(0))
+	assertDefaultValue(t, "KubeconfigBurst", defaultServerOptions.KubeconfigBurst, 0)
+	assertDefaultValue(t, "AuthenticationAudience", defaultServerOptions.AuthenticationAudience, "")
+	assertDefaultValue(t, "ProxyStrategies", defaultServerOptions.ProxyStrategies, "default")
+	assertDefaultValue(t, "CipherSuites", defaultServerOptions.CipherSuites, "")
+}
+
+func assertDefaultValue(t *testing.T, fieldName string, actual, expected interface{}) {
+	t.Helper()
+	actualType := reflect.TypeOf(actual)
+	expectedType := reflect.TypeOf(expected)
+	if actualType != expectedType {
+		t.Fatalf("For field %s, expected a value of type %s but got %s.", fieldName, actualType, expectedType)
+	}
+	if actual != expected {
+		t.Fatalf("For field %s, expected a value %v but got %v.", fieldName, actual, expected)
+	}
+}
+
+func TestValidateServerPortRange(t *testing.T) {
```

Comment:
(optional) consider combining these 3 port validation tests into a single table-driven test.

Reply:
Switched to a generic table driven test.
Not convinced its an improvement but ....